### PR TITLE
Logging tensorboard consistency

### LINKED ISSUE #993: Imagenet, MNIST, and Cifar10: consistency with Tensorboard and logging
Since these models are pretty similar, we should try to make them consistent with what they log to stdout and to Tensorboard.

Added Tensorboard support for imagenet in https://github.com/pytorch/xla/pull/985 and we should add the same functionality for MNIST and Cifar10

### PR DESCRIPTION
**Pull request by zcain117**
fixes #993 

Tested: ran all 5 affected training scripts both with and without `logdir` to make sure that tensorboard summaries are correctly written

### CODE REVIEW

File: test/test_train_mp_mnist.py [hunk 1]
```diff
@@ -137,9 +140,12 @@ def test_loop_fn(loader):
   for epoch in range(1, FLAGS.num_epochs + 1):
     para_loader = pl.ParallelLoader(train_loader, [device])
     train_loop_fn(para_loader.per_device_loader(device))
+    if xm.is_master_ordinal():
+      print("Finished training epoch {}".format(epoch))
```
Comment (taylanbil) [inline test/test_train_mp_mnist.py]: you can use `xm.master_print`

Comment (zcain117) [reply test/test_train_mp_mnist.py]: Good call, thanks

File: test/test_train_mnist.py [hunk 2]
```diff
@@ -163,6 +166,8 @@ def test_loop_fn(model, loader, device, context):
     if FLAGS.metrics_debug:
       print(met.metrics_report())
 
+  if writer:
```
Comment (dlibenzi) [inline test/test_train_mnist.py]: This seems pretty ugly. Are we sure we need to do this?

Comment (zcain117) [reply test/test_train_mnist.py]: probably not strictly necessary for imagenet since the data auto-flushes every 120sec, but for mnist and cifar the summary will be missing data unless we do this:
https://github.com/pytorch/pytorch/issues/24234

Comment (zcain117) [reply test/test_train_mnist.py]: Switched to a util method

**Comment by dlibenzi** [comment]
Let's make sure Facebook is OK in installing tensorboard.
Seems like an heavy dependency for me, but not making the call.

**Comment by zcain117** [comment]
> Let's make sure Facebook is OK in installing tensorboard.
> Seems like an heavy dependency for me, but not making the call.

@ailzhang seemed OK with installing it for the circleci test but she thought we might also need to install it for pytorch tests. Another option is just to not offer tensorboard summaries for `test_train_mnist` and `test_train_cifar`. They are pretty simple models and I don't know how useful it is to have the accuracy graph.

**Comment by dlibenzi** [comment]
Why not having?
```Python
def create_writer(path):
  if path is not None:
    import tensorboard
    ...
    return ...
  return None
```

**Comment by zcain117** [comment]
> Yea in pytorch we run mnist test at least. https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L183-L192
> I can submit a PR to install tensorboard there for test if necessary. We just need to coordinate the land of two PRs to minimize breakage time. :D

Do we pass --logdir in the pytorch test? I'll make the change that Davide suggested and then we won't need to install tensorboard for the tests

**Comment by ailzhang** [comment]
@zcain117 which --logdir are you referring to? (I think we probably don't but easy to add). 
Also are we going to monitor tensorboard output somewhere?

**Comment by zcain117** [comment]
> @zcain117 which --logdir are you referring to? (I think we probably don't but easy to add).
> Also are we going to monitor tensorboard output somewhere?

@ailzhang I was hoping we don't pass it in the tests. I'm switching to Davide's suggestion of only importing tensorboard If we supply any string via `--logdir`. That way, we won't import it for tests but it will work for users if they want to use it. I think automated test for tensorboard output is overkill and might break if tensorboard changes their format, but I'm open to other opinions. I was just going to manually test

**Comment by ailzhang** [comment]
Oh I see, if tensorboard is for users, I agree that it's not a hard deps, Davide's suggestion will work better :D 
(I thought we were going to have circleci results display somewhere :P

**Comment by zcain117** [comment]
Ran all 5 test files with and without `--logdir` as manual testing. Removed the `tensorboard` dependency from circleci since the tests should now pass without it. I'll wait for circleci to finish before submitting
